src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskType;
+import duke.task.ToDo;
+
+public class Parser {
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK_DONE = "done";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_NEW_TODO = "todo";
+    private static final String COMMAND_NEW_DEADLINE = "deadline";
+    private static final String COMMAND_NEW_EVENT = "event";
+
+    /**
+     * Takes in the userInput String and returns the corresponding command to be executed
+     *
+     * @param userInput The String that contains the user's input
+     * @return The corresponding Command based on the user's input
+     * @throws DukeException If the userInput String is not in the correct format or
+     *                       does not correspond to a valid command
+     */
+    public static Command parse(String userInput) throws DukeException {
+        if (userInput.equals(COMMAND_EXIT)) {
+            return new ExitCommand();
+        }
+        if (userInput.equals(COMMAND_LIST)) {
+            return new ListCommand();
+        }
+        if (userInput.startsWith(COMMAND_MARK_DONE)) {
+            int taskIndex = -1;
+            try {
+                taskIndex = Integer.parseInt(userInput.substring(5).strip()) - 1;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please provide a valid number.");
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("Please specify the index of the task.");
+            }
+            return new DoneCommand(taskIndex);
+        }
+        if (userInput.startsWith(COMMAND_DELETE)) {
+            int taskIndex = -1;
+            try {
+                taskIndex = Integer.parseInt(userInput.substring(7).strip()) - 1;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please provide a valid number.");
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("Please specify the index of the task.");
+            }
+            return new DeleteCommand(taskIndex);
+        }
+        if (userInput.startsWith(COMMAND_NEW_TODO)) {
+            Task task = parseTask(userInput, TaskType.TODO);
+            return new AddCommand(task);
+        }
+        if (userInput.startsWith(COMMAND_NEW_DEADLINE)) {
+            Task task = parseTask(userInput, TaskType.DEADLINE);
+            return new AddCommand(task);
+        }
+        if (userInput.startsWith(COMMAND_NEW_EVENT)) {
+            Task task = parseTask(userInput, TaskType.EVENT);
+            return new AddCommand(task);
+        }
+        throw new DukeException("Sorry, I do not understand your command.");
+    }
+
+    /**
+     * Takes in the userInput String and returns a Task that is to be added to taskManager
```

Comment:
Well done here! You have provided javadocs comments across the Java files in a consistent fashion.